src/map/clif.cpp

```diff
@@ -4671,17 +4671,14 @@ void clif_tradeadditem( struct map_session_data* sd, struct map_session_data* ts
 ///     0 = success
 ///     1 = overweight
 ///     2 = trade canceled
-void clif_tradeitemok(struct map_session_data* sd, int index, int fail)
+void clif_tradeitemok(struct map_session_data& sd, int index, e_exitem_add_result result)
 {
-	int fd;
-	nullpo_retv(sd);
+	PACKET_ZC_ACK_ADD_EXCHANGE_ITEM* p = reinterpret_cast<PACKET_ZC_ACK_ADD_EXCHANGE_ITEM*>(packet_buffer);
```

Comment:
```suggestion
	PACKET_ZC_ACK_ADD_EXCHANGE_ITEM p = {};
```
The pointer/packet_buffer variant should only be used for dynamic packets.